hal/common/retarget.cpp

```diff
@@ -138,14 +138,21 @@ static inline int openmode_to_posix(int openmode) {
 #endif
     return posix;
 }
+#if defined(__MICROLIB)
+extern "C" WEAK void mbed_sdk_init(void);
```

Comment:
Indeed, I just need to move it earlier into the file to cover microlib case - doing the change